runelite-client/src/main/java/net/runelite/client/plugins/gpu/GpuPlugin.java

```diff
@@ -879,10 +807,27 @@ else if (lastStretchedDimensions != null)
 			}
 
 			final Texture[] textures = textureProvider.getTextures();
-			final int heightOff = client.getViewportYOffset();
-			final int widthOff = client.getViewportXOffset();
+			int renderHeightOff = client.getViewportYOffset();
+			int renderWidthOff = client.getViewportXOffset();
+			int renderCanvasHeight = canvasHeight;
+			int renderViewportHeight = viewportHeight;
+			int renderViewportWidth = viewportWidth;
 
-			gl.glViewport(widthOff, canvasHeight - viewportHeight - heightOff, viewportWidth, viewportHeight);
+			if (client.isStretchedEnabled())
+			{
+				Dimension dim = client.getStretchedDimensions();
+				renderCanvasHeight = dim.height;
+
+				float scaleFactorY = dim.height / (float) canvasHeight;
```

Comment:
Use dim.getHeight() instead of dim.height and you won't have to cast canvasHeight to float.

Reply:
Good idea, changed.